articles/azure-functions/functions-reference-powershell.md

```diff
@@ -0,0 +1,424 @@
+    StatusCode = $status
+    Body = "Hello $name!"
+})
+```
+
+The result of invoking this function would be:
+
+```
+PS > irm http://localhost:5001?Name=Functions
+Hello Functions!
+```
+
+## PowerShell version
+
+The following table shows the PowerShell version used by each major version of the Functions runtime:
+
+| Functions version | PowerShell version                             |
+|-------------------|------------------------------------------------|
+| 1.x               | Windows PowerShell 5.1 (locked by the runtime) |
+| 2.x               | PowerShell Core 6.1                            |
+
+You can see the current version by printing `$PSVersionTable` from any function.
+
+## Dependency management
+
+Leveraging your own custom modules or modules from the [PowerShell Gallery](https://powershellgallery.com) is a little different than how you would do it normally.
+
+When you install the module on your local machine, it goes in one of the globally available folders in your `$env:PSModulePath`. Since your function will be running in Azure, you won't have access to the modules installed on your machine so the `$env:PSModulePath` for a PowerShell Function App is different than that of regular PowerShell.
+
+A PowerShell Function App's `PSModulePath` contains two paths:
+
+1. A `Modules` folder that exists at the root of your Function App
+1. A path to a `Modules` folder that lives inside the PowerShell language worker
+
+### Function App level `Modules` folder
+
+In order to leverage custom modules or PowerShell modules from the PowerShell Gallery (via `Save-Module`), you can place modules that your function depends on in the well-known `Modules` folder and they will be automatically available for you inside the Functions runtime.
+
+To take advantage of this, from the root of your _Function App_, create a `Modules` folder if it doesn't already exist and save the module you want to it.
+
+```powershell
+mkdir ./Modules
+Save-Module MyGalleryModule -Path ./Modules
+```
+
+>[!Note]
+> Remember, your Function App is the container of _all_ of you're functions for a particular project.
+
+Use `Save-Module` to save all of the modules you depend on or copy your own custom modules to the `Modules` folder. Your folder structure should look like this:
+
+```
+PSFunctionApp
+ | - MyFunction
+ | | - run.ps1
+ | | - function.json
+ | - Modules
+ | | - MyGalleryModule
+ | | - MyOtherGalleryModule
+ | | - MyCustomModule.psm1
+ | - local.settings.json
+ | - host.json
+```
+
+When you start your Function App, the PowerShell language worker will add this `Modules` folder to the `$env:PSModulePath` so that you can rely on module auto-loading just as you would in a regular PowerShell script.
+
+### Language worker level `Modules` folder
+
+We ship a few commonly used modules in the last position of the `PSModulePath`. The current list of modules are:
+
+- [PowerShellGet](https://www.powershellgallery.com/packages/powershellget)
```

Comment:
Perhaps we shouldn't include PSGet by default so that users `Save-Module` to their project rather than installing at runtime

Reply:
This was kept in for `Publish-Module` also, there is documentation in this article on the use of `Install-Module`

Reply:
Based on our discussion, I would suggest leaving PSGet in the image, but remove from the documentation.

Reply:
Removed